File: src/python/coingecko.py
import requests

# CoinGecko API URL
url = 'https://api.coingecko.com/api/v3/simple/price'
btc_params = {
    'ids': 'bitcoin',
    'vs_currencies': 'usd'
}

eth_params = {
    'ids': 'ethereum',
    'vs_currencies': 'usd'
}

sol_parmas = {
    'ids': 'solana',
    'vs_currencies': 'usd'
}


response = requests.get(url, params=btc_params)
data = response.json()

# 打印比特幣價格
print('Bitcoin price (USD):', data['bitcoin']['usd'])
# print('Ethereum price (USD):', data['ethereum']['usd'])
# print('Solana price (USD):', data['solana']['usd'])

File: src/coingecko.rs
use reqwest;

pub async fn get_coingecko() -> Result<(), reqwest::Error> {
    let response = reqwest::get("https://www.coingecko.com").await?; // Await the get function call and handle the Result properly
    println!("{:?}", response);
    Ok(())
}

pub async fn gecko_get_price(coin:String) -> Result<String, reqwest::Error> { // Change the return type to Result<String, reqwest::Error>
    let response = reqwest::get("https://api.coingecko.com/api/v3/simple/price?ids=bitcoin&vs_currencies=usd").await?; // Await the get function call and handle the Result properly
    let price = response.text().await?; // Await the text function call and handle the Result properly
    Ok(price)
}